--- Gruntfile.js	2016-12-22 10:02:55.763233997 -0500
+++ new.js	2016-12-22 10:02:00.083913992 -0500
@@ -112,6 +112,10 @@
 			]
 		},
 
+		exec: {
+		  genindex: '/generate-index.py --config /slides/config.json --template /pres/index.template --output /reveal.js/index.html /slides'
+		},
+
 		watch: {
 			js: {
 				files: [ 'Gruntfile.js', 'js/reveal.js' ],
@@ -131,8 +135,15 @@
 			markdown: {
 				files: [ '*.md' ]
 			},
+			config: {
+				files: ['config.json', 'index.template'],
+				tasks: 'genindex',
+				options: {
+					atBegin: true
+				}
+			},
 			options: {
-				livereload: true
+				livereload: true,
 			}
 		}
 
@@ -148,6 +159,7 @@
 	grunt.loadNpmTasks( 'grunt-contrib-connect' );
 	grunt.loadNpmTasks( 'grunt-autoprefixer' );
 	grunt.loadNpmTasks( 'grunt-zip' );
+	grunt.loadNpmTasks( 'grunt-exec');
 
 	// Default task
 	grunt.registerTask( 'default', [ 'css', 'js' ] );
@@ -173,4 +185,6 @@
 	// Run tests
 	grunt.registerTask( 'test', [ 'jshint', 'qunit' ] );
 
+  // Rebuild index.html
+	grunt.registerTask( 'genindex', [ 'exec:genindex'] );
 };
