diff --git a/Gruntfile.js b/Gruntfile.js
index 87630d5..687d5cd 100644
--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -101,7 +101,7 @@ module.exports = function(grunt) {
 		},
 
 		zip: {
-			'reveal-js-presentation.zip': [
+			'/reveal-js-presentation.zip': [
 				'index.html',
 				'css/**',
 				'js/**',
@@ -112,6 +112,10 @@ module.exports = function(grunt) {
 			]
 		},
 
+		exec: {
+		  genindex: '/generate-index.py --config /slides/config.json --template /pres/index.template --output /reveal.js/index.html /slides'
+		},
+
 		watch: {
 			js: {
 				files: [ 'Gruntfile.js', 'js/reveal.js' ],
@@ -131,8 +135,15 @@ module.exports = function(grunt) {
 			markdown: {
 				files: [ '*.md' ]
 			},
+			config: {
+				files: ['config.json', 'index.template'],
+				tasks: 'genindex',
+				options: {
+					atBegin: true
+				}
+			},
 			options: {
-				livereload: true
+				livereload: true,
 			}
 		}
 
@@ -148,6 +159,7 @@ module.exports = function(grunt) {
 	grunt.loadNpmTasks( 'grunt-contrib-connect' );
 	grunt.loadNpmTasks( 'grunt-autoprefixer' );
 	grunt.loadNpmTasks( 'grunt-zip' );
+	grunt.loadNpmTasks( 'grunt-exec');
 
 	// Default task
 	grunt.registerTask( 'default', [ 'css', 'js' ] );
@@ -173,4 +185,6 @@ module.exports = function(grunt) {
 	// Run tests
 	grunt.registerTask( 'test', [ 'jshint', 'qunit' ] );
 
+  // Rebuild index.html
+	grunt.registerTask( 'genindex', [ 'exec:genindex'] );
 };
