--- original.js	2016-09-21 08:42:52.670831499 -0400
+++ markdown.js	2016-09-21 08:39:38.241241559 -0400
@@ -21,14 +21,27 @@
 		throw 'The reveal.js Markdown plugin requires marked to be loaded';
 	}
 
+	var markedOptions = {};
+
+	if(typeof mermaid !== 'undefined' ) {
+	    markedOptions.renderer = new marked.Renderer();
+	    markedOptions.renderer.code = function(code, language){
+	        if(code.match(/^sequenceDiagram/)||code.match(/^graph/)){
+	            return '<div class="mermaid">'+code+'</div>';
+	        } else {
+	            return '<pre><code>'+code+'</code></pre>';
+	        }
+	    };
+	}
+
 	if( typeof hljs !== 'undefined' ) {
-		marked.setOptions({
-			highlight: function( lang, code ) {
+		marked.highlight = function( lang, code ) {
 				return hljs.highlightAuto( lang, code ).value;
-			}
-		});
+		}
 	}
 
+	marked.setOptions(markedOptions);
+
 	var DEFAULT_SLIDE_SEPARATOR = '^\r?\n---\r?\n$',
 		DEFAULT_NOTES_SEPARATOR = 'note:',
 		DEFAULT_ELEMENT_ATTRIBUTES_SEPARATOR = '\\\.element\\\s*?(.+?)$',
@@ -381,6 +394,10 @@
 					section.appendChild( notes );
 				}
 
+				// Call mermaid
+				if(typeof mermaid !== 'undefined' ) {
+					mermaid.initialize({ cloneCssStyles: false });
+				}
 			}
 
 		}
