--- markdown.js	2016-12-13 15:43:00.131472176 -0500
+++ markdown_new.js	2016-12-13 15:30:04.393206132 -0500
@@ -21,14 +21,32 @@
 		throw 'The reveal.js Markdown plugin requires marked to be loaded';
 	}

+  var markedOptions = {};
+	if(typeof Viz !== 'undefined' ) {
+    markedOptions.renderer = new marked.Renderer();
+    markedOptions.renderer.code = function(code, language){
+	        if(language === "graph") {
+						try {
+							var svg = Viz(code, options={ format:"svg", engine:"dot", scale:1 });
+							return svg;
+						}
+						catch(err) {
+							return '<pre><code class="no-highlight">'+err+'</code></pre>';
+						}
+	        } else {
+	            return '<pre><code class="'+language+'">'+code+'</code></pre>';
+	        }
+	    };
+	}
+
 	if( typeof hljs !== 'undefined' ) {
-		marked.setOptions({
-			highlight: function( lang, code ) {
+		marked.highlight = function( lang, code ) {
 				return hljs.highlightAuto( lang, code ).value;
-			}
-		});
+		}
 	}

+	marked.setOptions(markedOptions);
+
 	var DEFAULT_SLIDE_SEPARATOR = '^\r?\n---\r?\n$',
 		DEFAULT_NOTES_SEPARATOR = 'note:',
 		DEFAULT_ELEMENT_ATTRIBUTES_SEPARATOR = '\\\.element\\\s*?(.+?)$',
