--- mermaid.js	2016-09-21 08:46:32.984366913 -0400
+++ mermaid-new.js	2016-09-21 08:47:05.771297779 -0400
@@ -58455,7 +58455,7 @@
     /**
      * **cloneCssStyles** - This options controls whether or not the css rules should be copied into the generated svg
      */
-    cloneCssStyles: true,
+    cloneCssStyles: false,
 
     /**
      * **startOnLoad** - This options controls whether or mermaid starts when the page loads
